### PR TITLE
Allow custom connection string

### PR DESCRIPTION
**Pull request by Redsandro**
* Allow custom connection string
* Update `mongoose` for mongo:3.4 compatibility
* Update README.md

### CODE REVIEW

**Comment by Redsandro** [comment]
Yes, correct.

**Comment by mansona** [comment]
So I've just done a bit of an "update packages" release, updated Travis to run tests on Node 10 and added a Contributing guide.

As part of the contributing guide, I've added some scripts that start local docker images that would solve your problem of using a custom connection string https://github.com/stonecircle/express-autoroute-json/blob/master/CONTRIBUTING.md#running-tests

Does this solve your need so that you can run the tests locally? I would prefer not to add any complexity to the code so that it can run tests.

**Comment by Redsandro** [comment]
Hi Mansona, thanks for the review.

> Does this solve your need so that you can run the tests locally?

Not exactly. Port 27017 is semi-permanently in use here at the office, and other apps depend on it. For that reason, it's not a concept I particuar like to hardcode a dependency on a port (or a database connectionstring, for that matter).

I understand you prefer to not add complexity. But if I can have one shot at convincing you, we develop many Docker containers, and it's a mainstream concept to allow overrides with environment variables. If someone runs these tests in a container, or even just from a Makefile, really [this](https://github.com/stonecircle/express-autoroute-json/blob/74590f2d22a505c9227a6d277acc85de4c9eb9f1/test/index.js#L6) is the way to do it:

```
var dbConnectionString = process.env.DB_CONNECTION_STRING || 'mongodb://localhost/express-autoroute-json-test-database-piNMnJp8'
```

Basically this makes the tests 100% backwards compatible with the hardcoded connection you have set up for tests and CI, but allows for custom connection strings using an environment variable.

If you use CI or a Dockerized testing setup, Docker recommends using environment variables using the `-e` command line switch [or a `.env` file](https://docs.docker.com/compose/env-file).

**Comment by mansona** [comment]
Fixed by #34
